software/glasgow/protocol/bsdl.py

```diff
@@ -60,10 +60,16 @@ class BSDLDevice:
 
     scan_cells: list[BSDLScanCell]
 
+    idcode: bits | None = None
+
     @property
     def scan_length(self):
         return len(self.scan_cells)
 
+    @property
+    def idcode_length(self):
+        return 0 if self.idcode == 0 else len(self.idcode)
```

Comment:
Is `self.idcode == 0` ever true given its declared type? I'd expect this:

```suggestion
        return 0 if self.idcode is None else len(self.idcode)
```